pycytominer/tests/test_cyto_utils/test_sqlite.py

```diff
@@ -0,0 +1,426 @@
+""" Tests for pycytominer.cyto_utils.sqlite """
+
+import tempfile
+
+import pytest
+from pycytominer.cyto_utils.sqlite import (
+    LIKE_NULLS,
+    clean_like_nulls,
+    collect_columns,
+    contains_conflicting_aff_strg_class,
+    contains_str_like_null,
+    engine_from_str,
+    update_columns_to_nullable,
+    update_values_like_null_to_null,
+)
+from sqlalchemy import create_engine
+from sqlalchemy.engine.base import Engine
+from sqlalchemy.exc import IntegrityError
+
+
+@pytest.fixture
+def database_engine_for_testing() -> Engine:
+    """
+    A database engine for testing as a fixture to be passed
+    to other tests within this file.
+    """
+
+    # get temporary directory
+    tmpdir = tempfile.gettempdir()
+
+    # create a temporary sqlite connection
+    sql_path = "sqlite:///{}/test_sqlite.sqlite".format(tmpdir)
+    engine = create_engine(sql_path)
+
+    # statements for creating database with simple structure
+    create_stmts = [
+        """
+    drop table if exists tbl_a;
+    """,
+        """
+    create table tbl_a (
+    col_integer INTEGER NOT NULL
+    ,col_text TEXT
+    ,col_blob BLOB
+    ,col_real REAL
+    );
+    """,
+        """
+    drop table if exists tbl_b;
+    """,
+        """
+    create table tbl_b (
+    col_integer INTEGER
+    ,col_text TEXT
+    ,col_blob BLOB
+    ,col_real REAL
+    );
+    """,
+    ]
+
+    insert_vals = [1, "sample", b"sample_blob", 0.5]
```

Comment:
This notation is used for [bytes literals](https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals). It turned out to be a simple way to insert a blob into the test database tables and retain the blob storage class (or type) without needing to create or load a file object or similar. Inserting a string without the `b` notation results in a python string being inserted as a text storage class value. While this isn't crucial for testing with the current set, I felt it might be helpful to have in the future.